### DISCUSSION TITLE
trigger external camera events via mqtt

### DISCUSSION THREAD

**Discussion by aisbergde** [blakeblackshear/frigate#9409] [Ideas]
feature request: "external" objects, which can be triggered via mqtt and will just create snapshot and record for a specific duration (also to be defined via mqtt). A camera event and the related recording should be generated, mqtt events and all, what normally happens, when a normal object is detected.

Scnario:

- external motion sensors could be used to trigger events, force recordings and event notifications
- object detection of one camera could trigger parallel recording of other cameras, which did not detect the object
- in general HA scripts and automations could trigger recordings

a more advanced usage would be kind of:
- trigger an object "external person" only, if the camera did not detect an object "person".

This would allow to use external events only as a helper in case of camera detection issues.

Another modification could be: if the trigger event happens, the camera should use the box of the desired object type with the highest score, even it this score is below the threshold.

Or there could just be 2 different threshold values: one normal value and a second lower value, which will be used only a specific duration, triggered by an external event. the mqtt trigger could be: "use the lower threshold for x seconds"

**Comment by NickM-27** [comment]
This is already supported via the API in frigate 0.13

**Comment by aisbergde** [comment]
Regarding API - you mean what is described here:
https://deploy-preview-6262--frigate-docs.netlify.app/integrations/api#post-apieventscamera_namelabelcreate
Do we have some examples, how this could be used from inside homeassistant?
Currently, I am using some blueprints to send notifications to the HA App and to telegram. They use get methods, and I understand how to use. But I don't know how to use the post methods.

**Reply by NickM-27** [reply]
some users have created a `rest_command` inside of HA which then creates a service that can be used. In a future version of the frigate integration it will be provided automatically
